flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/api/output/BoundedStreamTask.java

```diff
@@ -106,7 +106,8 @@ protected void processInput(MailboxDefaultAction.Controller controller) throws E
             mainOperator.processElement(streamRecord);
         } else {
             mainOperator.endInput();
-            controller.allActionsCompleted();
+            controller.suspendDefaultAction();
```

Comment:
is it necessary to suspend the default action? Or is it just a precaution?

Reply:
I think we might have to suspend the default action, otherwise after resuming the mailbox processor would not blocked  on waiting for the next mail in `processMailsWhenDefaultActionUnavailable`, instead it would repeat the loop of check mail exists & run default action, which would be similar to a `while` loop~

Reply:
I think what Yun wrote makes sense.